discovery/wh_discovery.go

```diff
@@ -96,7 +96,11 @@ func (w *webhookPool) GetOrchestrators(numOrchestrators int) ([]*net.Orchestrato
 		return nil, err
 	}
 
-	return w.pool.GetOrchestrators(numOrchestrators)
+	w.mu.Lock()
```

Comment:
I think a RLock would be okay here. Also, a `defer` might be cleaner, since there's no need for an intermediate variable:

```
w.mu.RLock()
defer w.mu.RUnLock()
return w.pool.GetOrchestrators
```

Reply:
Good idea!